sycl/test-e2e/Compat/helloworld.cpp

```diff
@@ -0,0 +1,136 @@
+/***************************************************************************
+ *
+ *  Copyright (C) Codeplay Software Ltd.
+ *
+ *  Part of the LLVM Project, under the Apache License v2.0 with LLVM
+ *  Exceptions. See https://llvm.org/LICENSE.txt for license information.
+ *  SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ *  SYCLcompat API
+ *
+ *  helloworld.cpp
+ *
+ *  Description:
+ *    Checks that the SYCLcompat example program compiles and runs
+ **************************************************************************/
+
+// RUN: %{build} -o %t.out
+// RUN: %{run} %t.out
+
+#include <sycl/sycl.hpp>
+#include <syclcompat/syclcompat.hpp>
+
+#include <cassert>
+#include <iostream>
+
+#include <sycl/sycl.hpp>
```

Comment:
`sycl/sycl.hpp` is included twice, should only be once.

Also the current guidelines are to include as little as needed in E2E tests, so not the full `sycl.hpp` and `syclcompat.hpp`, in order to reduce testing times. Though I guess this particular test could be an exception if it's supposed to be used as learning material.

Reply:
I rather keep the full headers on this one since it will be used as an example.